python/tskit/drawing.py

```diff
@@ -488,9 +491,16 @@ def draw_x_axis(
                     self.add_text_in_group(
                         lab, tick, pos=(0, tick_length_lower), group_class="lab"
                     )
-            if sites is not None:
-                # Add sites as upper chevrons
-                for s in sites:
+            if site_ids is not None:
+                # Add sites as vertical lines with overlaid mutations as upper chevrons
+                site_muts = None
+                if hasattr(self, "node_mutations"):  # Make list of mutations to plot
```

Comment:
`self.node_mutations` is only assigned to an `SvgTree`. It isn't created (because it doesn't really make sense) within an instance of `SvgTreeSequence`.

I could set `self.node_mutations=None` in an `SvgTreeSequence`, but I think not creating it at all is more meaningful. Rather than `hasattr` I could alternatively put this in a try-except block.

Reply:
It's not a good sign, code shouldn't be testing for the presence or not of attributes regardless of how its done. I'll take a closer look at propose an alternative design when I get a chance.

Reply:
Point taken. I've just coded up a different approach. Will commit in a second

Reply:
https://github.com/tskit-dev/tskit/pull/1258/commits/69344d5730b06786891eb30c8a3bee17d36858c7